sympy/sets/tests/test_sets.py

```diff
@@ -1041,11 +1041,78 @@ def test_issue_8257():
     assert FiniteSet(-oo) + Interval(-oo, oo) == reals_plus_negativeinfinity
 
 
+def test_Union_imageset():
+    n = Dummy('n')
+    img1 = ImageSet(Lambda(n, 2 * n * pi), S.Integers)
+    assert Union(img1) == img1
+
+    assert Union(S.EmptySet, img1) == img1
+    assert Union(img1, S.EmptySet) == img1
+
+    img2 = ImageSet(Lambda(n, 2 * n * pi + pi), S.Integers)
+    assert Union(img1, img2) == ImageSet(Lambda(n, n * pi), S.Integers)
+
+    img2 = ImageSet(Lambda(n, 2 * n * pi + pi), Interval(0, 10))
+    uni = Union(img1, img2, evaluate=False)
+    assert Union(img1, img2) == uni
+
+    img2 = ImageSet(Lambda((n, m), 2 * n * pi + pi), S.Integers)
```

Comment:
ImageSet union is not implemented for multiple variable. this it the testcase for that. It will be returned as it is.